openmls/src/group/mls_group/proposal_store.rs

```diff
@@ -532,6 +533,139 @@ impl ProposalQueue {
         }
         Ok((proposal_queue, contains_own_updates))
     }
+
+    /// Filters received proposals without inline proposals
+    ///
+    /// 11.2 Commit
+    /// If there are multiple proposals that apply to the same leaf,
+    /// the committer chooses one and includes only that one in the Commit,
+    /// considering the rest invalid. The committer MUST prefer any Remove
+    /// received, or the most recent Update for the leaf if there are no
+    /// Removes. If there are multiple Add proposals for the same client,
+    /// the committer again chooses one to include and considers the rest
+    /// invalid.
+    ///
+    /// The function performs the following steps:
+    ///
+    /// - Extract Adds and filter for duplicates
+    /// - Build member list with chains: Updates & Removes
+    /// - Check for invalid indexes and drop proposal
+    /// - Check for presence of Removes and delete Updates
+    /// - Only keep the last Update
+    ///
+    /// Return a [`ProposalQueue`] and a bool that indicates whether Updates for
+    /// the own node were included
+    pub(crate) fn filter_proposals_without_inline(
```

Comment:
Is there a reason why this wouldn't consider inline proposals? I'd think it does.

Reply:
That function name is a bit wonky, yes. I copied the body from `filter_proposals`, which also takes an argument `inline_proposals`. The API of that function didn't quite work with the new commit builder, so I made a copy where I changed it. Part of that was removing the `inline_proposals` argument.  I didn't want to change the existing code because (a) we'll probably get rid of `create_commit` soon anyway and we can just remove the old version then and (b) it might just break things. If you think it's important, I can look into making `create_commit` use `filter_proposals_without_inline`. We could then remove the current `filter_proposals` and rename `filter_propoals_without_inline` to `filter_proposals`. It might work, but I am not sure (I forgot some details because I was away for a week)